File: src/com/prey/events/receivers/EventReceiver.java
package com.prey.events.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.prey.events.Event;
import com.prey.events.factories.EventFactory;
import com.prey.events.manager.EventManagerRunner;
import com.prey.events.manager.SignalFlareRunner;
 

public class EventReceiver extends BroadcastReceiver {

	@Override
	public void onReceive(Context context, Intent intent) {
		Event event=EventFactory.getEvent(context, intent);
		new Thread(new EventManagerRunner(context,event)).start();
		new Thread(new SignalFlareRunner(context,event)).start();
	}
	
}

File: app/src/main/java/com/prey/events/receivers/kotlin/EventReceiver.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.events.receivers.kotlin

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.prey.events.factories.kotlin.EventFactory
import com.prey.events.manager.kotlin.EventManagerRunner
import com.prey.kotlin.PreyConfig

class EventReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val isDeviceRegistered =
            PreyConfig.getInstance(context).isThisDeviceAlreadyRegisteredWithPrey()
        if (isDeviceRegistered) {
            val event = EventFactory.getEvent(context, intent)
            if (event != null) {
                Thread(EventManagerRunner(context, event)).start()
            }
        }
    }
}